### PR TITLE
Refactor/gh convention

### LINKED ISSUE #7: Sphinx can't import module BuildDoc
Running `make build `gives:
```

ModuleNotFoundError: No module named 'sphinx.setup_command'
make: *** [Makefile:24: docu] Error 1
```

We can simply remove the BuildDoc from setup.py and use setuptools only, or we can fix the import

### PR DESCRIPTION
**Pull request by JotaFan**
Example PR using proposed GH convention.

1. Linting conventions and application
2. Pyhton overall cleaup
3. Apply poetry to manage packaging
4. Small bug hunting? (to be edited)
5. Apply gh action?
6. Pre-commit?

Fixes #7

### CODE REVIEW

**Review by cedricdcc — APPROVED** [review]

**Review by marc-portier — COMMENTED** [review]
only remark for now is on the "what if poetry is already available"

File: Makefile [hunk 1]
```diff
@@ -15,10 +15,11 @@ clean:
 
 init:
 	pip install --upgrade pip
-	pip install -e .
+	pip install poetry
```
[the remark this answers is not in the document]
Comment (JotaFan) [reply Makefile:18]: done

**Review by cedricdcc — APPROVED** [review]

File: requirements.txt
Comment (bulricht) [inline requirements.txt:1]: do we keep this when pyproject.toml is present?

File: custom_cmds.py
Comment (bulricht) [inline custom_cmds.py:1]: these commands will probably become obsolete, as overlapping with poetry built-ins

**Review by bulricht — CHANGES_REQUESTED** [review]

File: requirements.txt
Comment (JotaFan) [reply requirements.txt:1]: well, no.
I kept them to have a easier way to just add or delete reqs. 
There is a make command to add the reqs from these files to poetry.

But i would say we can delete them (along with the make command)

File: Dockerfile
Comment (bulricht) [inline Dockerfile:1]: Guess we can drop this file as well, as we'll typically `pip install` this lib in other projects

File: Makefile [hunk 2]
```diff
@@ -1,7 +1,9 @@
-PYTHON = python
 TEST_PATH = ./tests/
 FLAKE8_EXCLUDE = venv,.venv,.eggs,.tox,.git,__pycache__,*.pyc
-SHELL := /bin/bash
+PROJECT = pykg2tbl
+AUTHOR = Marc Portier, Cedric Decruw 
+
+.PHONY: build docs format install jupyter publish run tests
```
Comment (bulricht) [inline Makefile:6]: phony keys don't match with the ones below: e.g. test vs tests, jupyter, ...

File: docs/source/modules.rst
Comment (bulricht) [inline docs/source/modules.rst:1]: wondering if we can generate this automatically from the Makefile, so we don't have to commit those .rst files

File: pykg2tbl/__main__.py [hunk 3]
```diff
@@ -256,8 +283,12 @@ def main(sysargs=None):
     vars_template = template_service.variables_in_query(args.template_name)
     if args.variables is not None and len(vars_template) > 0:
         params = args_values_to_params(args.variables)
-        variables_check(variables_template=vars_template, variables_given=params)
-    querry = template_service.build_sparql_query(name=args.template_name, variables=params)
+        variables_check(
+            variables_template=vars_template, variables_given=params
+        )
+    querry = template_service.build_sparql_query(
```
Comment (bulricht) [inline pykg2tbl/__main__.py:289]: typo: query instead of querry

File: pykg2tbl/service.py [hunk 4]
```diff
@@ -75,13 +75,18 @@ def __init__(self, *files):
         for f in files:
             log.debug(f"loading graph from file {f}")
             graph_to_add = g.parse(f)
-            self.graph = graph_to_add if self.graph is None else self.graph + graph_to_add
```
Comment (bulricht) [inline pykg2tbl/service.py:70]: what would happen if this line was just `self.graph += graph_to_add`

Comment (JotaFan) [reply pykg2tbl/service.py:70]: be better written. I am changing all these classes in another branch, so maybe leave that to a different PR

File: Makefile (hunk 2 above)
Comment (JotaFan) [reply Makefile:6]: we really should make a standard for these stuff

File: docs/source/modules.rst
Comment (JotaFan) [reply docs/source/modules.rst:1]: well they are, so, should we just drop these?

**Review by bulricht — APPROVED** [review]

**Pull request merged**